tpp-app/src/info/InfoService.js

```diff
@@ -0,0 +1,69 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { FACT_CYCLE } from '../services/utils/constants.js'
+import { getFullCurrentDateString } from "../services/utils/helpers.js"
+
+/**
+ * Retrieves an array containing the last date the user accessed the info screen and the fact number they saw
+ * @returns returns a promise resolving in an array with format ["2022-1-1", "1"]
+ */
+export const GETFactCycle = async () => new Promise(async (resolve, reject) => {
+    try {
+     if (await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM) == null) {
+        resolve(null);
+     } else {
+         value = await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM).then((value) => {
```

Comment:
not sure if this is your error but I don't think you should be using await and .then() together, just as try/except does the same job as .catch(). I would do it like this, try it and lmk if it fixes your error: 
```
export const GETFactCycle = async () => new Promise(async (resolve, reject) => {
    try {
     if (await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM) == null) {
        resolve(null);
     } else {
         value = await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM);
         console.log("Retrieved Fact Cycle Date and Number");
         resolve(value);
     }
    } catch (e) {
        console.log(`GETFactCycle error: ${JSON.stringify(e)}`)
        reject(null);
    }
});
```